Formula/b/bsc.rb

```diff
@@ -0,0 +1,109 @@
+class Bsc < Formula
+  desc "Bluespec Compiler (BSC)"
+  homepage "https://github.com/B-Lang-org/bsc"
+  url "https://github.com/B-Lang-org/bsc.git",
+    tag:      "2025.01.1",
+    revision: "65e3a87a17f6b9cf38cbb7b6ad7a4473f025c098"
+  license "BSD-3-Clause"
+  head "https://github.com/B-Lang-org/bsc.git", branch: "main"
+
+  depends_on "autoconf" => :build
+  depends_on "cabal-install" => :build
+  depends_on "ghc" => :build
+  depends_on "gperf" => :build
+  depends_on "make" => :build
+  depends_on "pkgconf" => :build
+  depends_on "gmp"
+  depends_on "icarus-verilog"
+  depends_on "tcl-tk@8"
+
+  uses_from_macos "bison" => :build
+  uses_from_macos "flex" => :build
+  uses_from_macos "libffi"
+
+  def install
+    system "cabal", "v2-update"
+    system "cabal", "v2-install", "--lib",
+                    "old-time",
+                    "regex-compat",
+                    "split",
+                    "syb"
+
+    store_dir = `cabal path --store-dir`.chomp
+    ghc_version = `ghc --numeric-version`.chomp
+    package_db = `echo #{store_dir}/ghc-#{ghc_version}*/package.db`.chomp
+
+    with_env(
+      PREFIX:           libexec,
+      GHCJOBS:          ENV.make_jobs.to_s,
+      GHCRTSFLAGS:      "+RTS -M4G -A128m -RTS",
+      GHC_PACKAGE_PATH: "#{package_db}:",
+    ) do
+      system "make", "install-src", "-j#{ENV.make_jobs}"
+    end
```

Comment:
Are we certain this is how `ENV.make_jobs` should be used?  As written, this will cause `make` to spawn that many parallel jobs, within each of which GHC will spawn the same number of parallel subjobs.  My first instinct is to balance this by taking the square root of that number and using that in both positions, but @chenrui333 may have better ideas.